dasharo-compatibility/check-ethernet-ports-order.robot

```diff
@@ -0,0 +1,71 @@
+*** Settings ***
+Library             Collections
+Library             DateTime
+Library             OperatingSystem
+Library             Process
+Library             String
+Library             Telnet    timeout=20 seconds
+Library             SSHLibrary    timeout=90 seconds
+Library             RequestsLibrary
+Resource            ../rtectrl-rest-api/rtectrl.robot
+Resource            ../variables.robot
+Resource            ../keywords.robot
+
+Suite Setup         Run Keywords
+...                     Prepare Test Suite
+Suite Teardown      Run Keyword
+...                     Log Out And Close Connection
+
+
+*** Test Cases ***
+SPS001.001 Ethernet ports are in order
+    [Documentation]    This test automates the verification of port order based
+    ...    on PCIe bus numbers and checks PCIe switching.
+    Power On
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Switch To Root User
+    ${pci_devices}=    Get MACs
+    Log    PCIe devices and their MACs: ${pci_devices}
+    Compare Interfaces    ${pci_devices}
+
+
+*** Keywords ***
+Get MACs
+    [Documentation]    Extract MAC addresses from lspci output.
+    ${lspci_output}=    Execute Command In Terminal    lspci -vvnn
+    ${lines}=    Split String    ${lspci_output}    \n
+    ${pci_devices}=    Create List
+
+    ${sfp_flag}=    Set Variable    ${FALSE}
+    FOR    ${line}    IN    @{lines}
+        IF    '[8086:125c]' in '${line}'
```

Comment:
This ID should be a variable settable from platform config. note that the controller IDs may change in the future (sometimes happens with new revisions of HW)

Reply:
moved to configs: https://github.com/Dasharo/open-source-firmware-validation/pull/515/commits/8b37fc9d8fc2d98ce6614a15bc9af8d905024f70